client/asset/eth/eth.go

```diff
@@ -580,36 +579,11 @@ func createWallet(createWalletParams *asset.CreateWalletParams, skipConnect bool
 		}
 
 		if !skipConnect {
-			ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
-
-			var unknownEndpoints []string
-
-			for _, endpoint := range endpoints {
-				known, compliant := providerIsCompliant(endpoint)
-				if known && !compliant {
-					return fmt.Errorf("provider %q is known to have an insufficient API for DEX", endpoint)
-				} else if !known {
-					unknownEndpoints = append(unknownEndpoints, endpoint)
-				}
-			}
-
-			if len(unknownEndpoints) > 0 && createWalletParams.Net == dex.Mainnet {
-				providers, err := connectProviders(ctx, unknownEndpoints, createWalletParams.Logger, big.NewInt(chainIDs[createWalletParams.Net]))
-				if err != nil {
-					return err
-				}
-				defer func() {
-					for _, p := range providers {
-						p.ec.Close()
-					}
-				}()
-				if len(providers) != n {
-					return fmt.Errorf("Could not connect to all providers")
-				}
-				if err := checkProvidersCompliance(ctx, walletDir, providers, createWalletParams.Logger); err != nil {
-					return err
-				}
+			if err := createAndCheckProviders(walletDir, endpoints,
+				createWalletParams.Net, createWalletParams.Logger, ctx); err != nil {
+				return fmt.Errorf("providers problem: %v", err)
```

Comment:
Maybe to be more specific: **providers problem** -> **create and check providers problem**

Reply:
Ok.